_posts/2016-07-23-migrating-a-heavily-used-government-service-to-aws.markdown

```diff
@@ -0,0 +1,105 @@
+
+The final savings calculation looked like this:
+
+* Old CDN/WAF provider: $16,000 per month
+* CloudFront distribution and OWASP-enabled proxy servers: $400 per month (including traffic costs)
+
+That's a **97.5%** savings we were delighted to return to the operating budget of the project for more important things like developing new features!
+
+In other blog posts we discuss in more detail the complexities of supporting [SSL certificates for our government customers's hostnames]({% post_url 2016-07-11-using-lets-encrypt-for-a-multi-tenant-web-application %}) and how we managed to comply with the [DNSSEC requirement]({% post_url 2016-07-09-making-dnssec-work-for-an-aws-hosted-saas %}) for government agencies. Those posts go into much more technical detail and are worth reading if you're interested in how we solve the security challenges of providing SaaS services for thousands of government agencies.
+
+## Challenges in the ATO Process
+
+The great thing about pools of servers built by hardened recipes is that individual server failures are a non-event: a new server is spun up automatically to replace one that failed. However, the Authority to Operate (ATO) process in particular and the government security auditing process in general become a bit tricky for an architecture like ours because these processes require (among many other things):
+
+* Vulnerability testing of individual servers
+* Compliance testing of individual servers
+* Penetration testing of individual servers and exposed pool endpoints
+
+Since individual servers are identified by IP address, these tests get a little bit more complicated when individual servers can be rebuilt without warning due to server failure, load spikes, or data center outages - because the new server will often come online with a different DHCP'd IP address than the one it replaced. Similarly, if a new server is spun up in response to increased load, it will have an IP address that is unknown to the security testing infrastructure and will cause unnecesssary red flags.
+
+We have worked closely with GSA security personnel to understand their security auditing requirements and to help them understand the new terrain in which we are operating. We hope this two-way conversation will lead to updated security practices that can take the dynamic nature of cloud hosting environments into account without sacrificing completeness or quality.
```

Comment:
I continue to love this potential outcome even though I never would have thought to mention it in a post!